### PR TITLE
Add a specific test for optional whitespace in HTTP headers

### PR DESCRIPTION
**Pull request by chrisvest**
Motivation:
Since we made header value validation more strict in 4.1.84, we've gotten the occasional issue reported that whitespace isn't handled correctly.

Modification:
Add a test that stresses the optional whitespace decoding of HTTP headers, because I think we didn't have one already.
Also include a javadoc comment with relevant RFC references.

Result:
This proves that we handle optional whitespace correctly when we decode HTTP headers.

### CODE REVIEW

**Review by normanmaurer — APPROVED** [review]

**Review by normanmaurer — CHANGES_REQUESTED** [review]
one nit.

File: codec-http/src/test/java/io/netty/handler/codec/http/HttpRequestDecoderTest.java [hunk 1]
```diff
@@ -602,6 +602,41 @@ public void testHttpMessageDecoderResult() {
...
+        assertTrue(channel.writeInbound(Unpooled.copiedBuffer(requestStr, CharsetUtil.US_ASCII)));
+        HttpRequest request = channel.readInbound();
+        assertTrue(request.decoderResult().isSuccess());
+        HttpHeaders headers = request.headers();
+        assertEquals("example.com", headers.get("Host"));
+        assertEquals("x0", headers.get("X-0-Header"));
+        assertEquals("x1", headers.get("X-1-Header"));
+        assertEquals("x2", headers.get("X-2-Header"));
+        assertEquals("x3", headers.get("X-3-Header"));
+        assertEquals(LastHttpContent.EMPTY_LAST_CONTENT, channel.readInbound());
```
Comment (normanmaurer) [inline codec-http/src/test/java/io/netty/handler/codec/http/HttpRequestDecoderTest.java:636]:
```suggestion
        LastHttpContent last = channel.readInbound();
        assertEquals(LastHttpContent.EMPTY_LAST_CONTENT, last);
        last.release();
```

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Comment by normanmaurer** [comment]
@chrisvest can you port to main as well ?
